### ISSUE TITLE
[Form] Possible bug in \Symfony\Component\Form\ResolvedFormType

### ISSUE THREAD

**Issue by c1tru55** [symfony/symfony#14919]
When I try to call `ResolvedFormType::getTypeExtensions()` I get an error:
> Attempted to call method "getExtensions" on class "Symfony\Component\Form\Extension\Core\Type\FormType".

As I see there are code for BC ( https://github.com/symfony/Form/blob/master/ResolvedFormType.php#L98 ):
```php
    public function getTypeExtensions()
    {
        // BC
        if ($this->innerType instanceof AbstractType) {
            return $this->innerType->getExtensions();
        }

        return $this->typeExtensions;
    }
```
but we don't have `use` statement for deprecated AbstractType:
```php
use Symfony\Component\Form\Exception\InvalidArgumentException;
use Symfony\Component\Form\Exception\UnexpectedTypeException;
use Symfony\Component\EventDispatcher\EventDispatcher;
use Symfony\Component\OptionsResolver\OptionsResolver;
```
so it is assumed that deprecated AbstractType class is stored in the same folder as ResolvedFormType class. But current AbstractType is also stored in the same folder with ResolvedFormType and it doesn't have `getExtensions()` method: https://github.com/symfony/Form/blob/master/AbstractType.php

**Comment by c1tru55** [comment]
https://github.com/symfony/Form/pull/15/files

**Comment by jakzal** [comment]
@c1tru55 please submit PRs to the `symfony/symfony` repository, instead of specific component repositories (which are just mirrors).

**Comment by c1tru55** [comment]
done #14921
what about other 2.x branches?

**Comment by jakzal** [comment]
@c1tru55 you should submit your PR to the latest supported branch that you can reproduce this bug on. We merge older branches to the newer ones regularly.

**Comment by c1tru55** [comment]
@jakzal it seems that this bug exists in all branches starting from 2.3, so I made PR to 2.8

**Comment by phansys** [comment]
@c1tru55, check https://symfony.com/doc/current/contributing/code/patches.html#choose-the-right-branch.

**Comment by c1tru55** [comment]
@jakzal what about this: #14922 ?

**Comment by jakzal** [comment]
2.4 is not supported. I think the bug is still present in 2.3, no?

 However, even if you send a PR to 2.8, we can still merge it to 2.3. So don't worry about it too much.

**Comment by Tobion** [comment]
This is already fixed #14896

**Comment by phansys** [comment]
@c1tru55, just for future reference, if you can reproduce a bug in [2.3](https://github.com/symfony/symfony/tree/2.3), you should make your PR against 2.3 (since 2.3 is the older maintained branch where that bug exists).

**Comment by c1tru55** [comment]
@phansys this bug is fixed in 2.3, but appeared again in 2.4-2.x. So in what branch I should make a PR?

**Comment by phansys** [comment]
2.4 has reached EOM: https://symfony.com/roadmap.json?version=2.4
So, I think it should be done against 2.6 (which is the older branch with EOM not yet reached).
Please check also https://symfony.com/doc/current/contributing/community/releases.html#schedule.

**Comment by jakzal** [comment]
@phansys @c1tru55 it's already fixed in 2.3 as Tobion mentioned. Soon it'll be merged into other branches.

**Comment by phansys** [comment]
Thanks @jakzal, @Tobion.
